unlock-app/src/middlewares/storageMiddleware.js

```diff
@@ -158,6 +176,38 @@ const storageMiddleware = config => {
       dispatch(setError(Storage.Warning('Could not initiate key purchase.')))
     })
 
+    const { router } = getState()
+    if (router && router.location && router.location.pathname === '/recover/') {
```

Comment:
Would it be better to trigger this from the recover page with an action instead of parsing the router here? That might make it more flexible for unanticipated future use, but I'm ok with it either way.

Reply:
That's always something that I am not sure about: how to dispatch actions without user input. We could use some kind of lifecycle method (like after rendering) but that fells gross.